Configuration API quickstart docs (#3009)
dapr/docs: daprdocs/content/en/developing-applications/building-blocks/configuration/configuration-api-overview.md

description: "Overview of the configuration API building block"
---

{{% alert title="Preview" color="primary" %}}
  This API is currently in `Alpha` state.
{{% /alert %}}

Consuming application configuration is a common task when writing applications. Frequently, configuration stores are used to manage this configuration data. A configuration item is often dynamic in nature and tightly coupled to the needs of the application that consumes it. 

For example, application configuration can include:
- Names of secrets
- Different identifiers
- Partition or consumer IDs
- Names of databases to connect to, etc 

> Usually, configuration items are stored as key/value items in a state store or database. Developers or operators can change application configuration at runtime. Once changes are made, the developer will be notified to take the required action and load the new configuration. 

Reviewer: ```suggestion
Usually, configuration items are stored as key/value items in a state store or database. Developers or operators can change application configuration at runtime in the configuration store. Once changes are made, a service is notified to load the new configuration. 
```

Configuration data is read-only from the application API perspective, with updates to the configuration store made through operator tooling. With Dapr's configuration API, you can:
- Consume configuration items that are returned as read-only key/value pairs
- Subscribe to changes whenever a configuration item changes

<img src="/images/configuration-api-overview.png" width=900>

{{% alert title="Note" color="primary" %}}
 The Configuration API should not be confused with the [Dapr sidecar and control plane configuration]({{< ref "configuration-overview" >}}), which is used to set policies and settings on Dapr sidecar instances or the installed Dapr control plane.
{{% /alert %}}

## Try out configuration

### Quickstart

> Want to put the Dapr pub/sub API to the test? Walk through the following quickstart and tutorials to see pub/sub in action:

Reviewer 2: ```suggestion
Want to put the Dapr configuration API to the test? Walk through the following quickstart to see configuration API in action:
```

> Want to put the Dapr configuration API to the test? Walk through the following quickstart and tutorials to see the configuration API in action:

Reviewer: ```suggestion
Want to put the Dapr configuration API to the test? Walk through the following quickstart to see the configuration API in action:
```

| Quickstart | Description |
| ---------- | ----------- |

> | [Configuration quickstart]({{< ref configuration-quickstart.md >}}) | Send and receive messages using the publish and subscribe API. |

Reviewer 2: ```suggestion
| [Configuration quickstart]({{< ref configuration-quickstart.md >}}) | Get configuration items or subscribe to configuration changes using the configuration API. |
```
